### PR TITLE
Implement API pagination

### PR DESCRIPTION
**Pull request by schi314**
APIs can now paginate responses (that MUST be an array) into subsets.

For #165 

Example usage:
1. Type cast the response as `PaginateableResponse` in your endpoint: `const res  = expressRes as PaginateableResponse;`
2. Use the `res` object as usual
3. Use `res.status(200).paginate(AnArrayOfEntries)` for paginating your response

Testing:
* Integration tests have been made for `GET /encounters` and `GET /persons`
* Tests have been passed

### CODE REVIEW

File: backend/src/routes/__test__/person.route.test.ts [hunk 1]
```diff
@@ -57,13 +132,365 @@ describe('person ', () => {
...
 
     // retrieve encounter from database, and check that the updated encounter contains the changed location
     const resPerson = await supertest(app).get(`/api/persons/${newPersonId}`);
     expect(resPerson._body.location).toBe(requestPersonData.location);
   });
 });
+
+describe('GET /persons', () => {
+  it('Response paginated and returns correct number of entries', async () => {
+    const person1 = new Person(person1Data);
```
Comment (AdamAWiener) [inline backend/src/routes/__test__/person.route.test.ts:148]: Same comment as on the encounter tests, could we extract this to a method?

**Review by AdamAWiener — CHANGES_REQUESTED** [review]
Looks nice - good change to include pagination. Just a refactoring change in the test files!

File: backend/src/routes/__test__/encounter.route.test.ts [hunk 2]
```diff
@@ -411,3 +419,396 @@ describe('encounter ', () => {
...
+            .set('Authorization', token)
+            .query({
+                page: 4
+            });
+
+        expect(encounters.length).toEqual(10);
+    });
+
+    it('Response not paginated when page is not given', async () => {
+        const encounter1 = new Encounter(encounter1Data);
```
[the remark this answers is not in the document]
Comment (schi314) [reply backend/src/routes/__test__/encounter.route.test.ts]: Yep, will do

File: backend/src/routes/__test__/person.route.test.ts (hunk 1 above)
Comment (schi314) [reply backend/src/routes/__test__/person.route.test.ts:149]: Yep, will do

**Review by sher812 — APPROVED** [review]
LGTM

**Review by sdiv877 — APPROVED** [review]
LGTM! Thanks for adding this!

**Pull request merged**